### ISSUE TITLE
v0.35.0 does not work on Intel Macs

### ISSUE THREAD

**Issue by core-code** [ipfs/ipfs-desktop#2809]
v0.35.0 is ARM/"AppleSilicon" only, and cannot be launched on Intel Macs:

<img width="252" alt="Screenshot 2024-05-14 at 12 57 47" src="https://github.com/ipfs/ipfs-desktop/assets/1261954/97d641d0-0993-47b6-8195-ed997e4d2083">

**Comment by jc-frosty** [comment]
TLDR: When compiled yourself on such a system it runs.
I updated the application using its built in update function and ran into this issue.  I thought this might be an issue with compiling or packaging and so I built it on my machine from source and the application compiled and built fine, it just has to be run from the directory that I built it in using the command `npm start` in a cli environment.  ifs desktop just runs under electron.
MacOS Monterey
Mac Pro 6,1(intel)
IPFS desktop Revision: d137048
If future versions do become version or chip(apple silicon vs intel) specific, I would recommend implementing a step in the update process that checks whether the update is compatible with the system.

**Comment by sevenoh7** [comment]
Can you guide me how to build from source? I'm also having issues getting it to run on Monterey

**Comment by core-code** [comment]
should be addresses upstream instead of trying to build from source. @lidel ?

**Comment by jc-frosty** [comment]
Building from source is just a work around until the issue is fixed.

**Comment by jc-frosty** [comment]
> Can you guide me how to build from source? I'm also having issues getting it to run on Monterey

[Instructions](https://github.com/ipfs/ipfs-desktop/tree/main#install-from-source)

**Comment by d70-t** [comment]
One annoying thing about this issue is, that even if I downgrade to the previous version (which works fine), the auto-updater keeps breaking my installation after a while.

**Comment by lidel** [comment]
Thank you for reporting this and confirming local build works.

Seems something changed on Apple's side, or gitbub worker's side, because the only code difference between 0.34 and 0.35 is updating Kubo version:

https://github.com/ipfs/ipfs-desktop/compare/v0.34.0...v0.35.0

Will look into this.

### Temporary workaround

Until we have a bugfix release, a workaround is to downgrade to 0
automatic updates can be disabled via config:

https://github.com/ipfs/ipfs-desktop/blob/c89435c101fb9f5135465661c14197f583db3709/src/auto-updater/index.js#L13-L16

Find `config.json` in your `IPFS Desktop` app's profile directory and add `disableAutoUpdate` set to `true`:


```diff
{
	"ipfsConfig": {
		"path": "",
		"flags": [
			"--agent-version-suffix=desktop",
			"--migrate",
			"--enable-gc"
		]
	},
	"language": "",
	"experiments": {},
	"binaryPath": "",
	"__internal__": {
		"migrations": {
			"version": "0.35.0"
		}
	},
	"automaticGC": true,
+        "disableAutoUpdate": true,
	"openWebUIAtLaunch": true,
	"window": {
		"width": 1628,
		"height": 683
	}
}
```

If you do this, set a reminder to remove it at some point in the future, or you will be stuck with an old version that may get insecure over time.

**Comment by core-code** [comment]
the change is on github's side, they default to building on (and for) ARM now

**Comment by lidel** [comment]
Indeed, seems this caught a lot of projects by surprise:

https://github.com/actions/runner/issues/3256

Sounds like the most expedient fix is to downgrade mac builder to macos-13 image. Will give it a try and if that works, will ship a  patch release.

**Comment by lidel** [comment]
Shipped [v0.35.1](https://github.com/ipfs/ipfs-desktop/releases/tag/v0.35.1), hopefully solves the problem at hand, restoring Intel funcitonality.

Follow-up work is tracked in https://github.com/ipfs/ipfs-desktop/issues/2813

**Comment by core-code** [comment]
thanks, confirmed resolved.

**Comment by lidel** [comment]
Thanks for confirming.

Btw: since you have intel-based Mac, mind checking if the universal version linked [here](https://github.com/ipfs/ipfs-desktop/pull/1856#issuecomment-2116260076) also works on your machine? (comment on https://github.com/ipfs/ipfs-desktop/pull/1856#issuecomment-2116260076)

**Comment by core-code** [comment]
seems to launch fine here!
